engine/src/renderer.cpp

```diff
@@ -206,6 +225,79 @@ namespace TremorEngine {
 			#undef SlicesMax
 		}
 
+		// Draw object sprites
+		std::vector<Object *> *objects=getObjectsInRangeFunctor(camera, getObjectsInRangeUserData);
+
+		RendererCompareObjectsByDistance compareObjectsByDistance(camera);
+		std::sort(objects->begin(), objects->end(), compareObjectsByDistance); // sort so that we paint closer objects over the top of further away ones (the z buffer is not enough if textures are partially transparent)
+
+		for(auto object : *objects) {
+			// Determine angle from camera to object, and skip drawing if object is behind camera.
+			double dx=camera.getX()-object->getCamera().getX();
+			double dy=camera.getY()-object->getCamera().getY();
+			double objectAngleDirect=atan2(dy, dx);
+			double objectAngleRelative=angleNormalise(objectAngleDirect-camera.getYaw());
+			if (objectAngleRelative<0.5*M_PI || objectAngleRelative>1.5*M_PI)
```

Comment:
This kind of thing could probably be better as a reusable function.

Reply:
Good idea